File: 7KYU/Disarium Number (Special Numbers Series #3)/solution.cs
using System.Linq;
using System.Collections.Generic;
using System;
class Kata
{
    public static string DisariumNumber(int number)
{
  var sum = 0;
  var i = 1;
  foreach(var d in GetDigits(number))
  {
    sum += (int)Math.Pow(d, i);
    i++;
  }
  
  return (sum == number)
    ? "Disarium !!"
      : "Not !!";
}

public static IEnumerable<int> GetDigits(int number)
{
  var digits = new List<int>();
  
  while(number != 0)
  {
    digits.Add(number % 10);
    number = (int)Math.Floor((number / 10.0));
  }
  return digits.AsEnumerable().Reverse();
}
}

File: 7KYU/Disarium Number (Special Numbers Series #3)/solution.clj
(ns kata)

(defn number->digits [num]
  (->> num str (map (comp read-string str))))

(defn pow [n pow]
  (reduce * (repeat pow n)))

(defn disarium-number [number]
  (if (= number (as-> (number->digits number) o
                  (map (fn [n e] (pow n e)) o (range 1 (inc (count o))))
                  (reduce + o)))
    "Disarium !!"
    "Not !!"))